### DISCUSSION TITLE
Regarding RT-DETR finetuning

### DISCUSSION THREAD

**Discussion by arabinda-iitbbs** [ultralytics/ultralytics#13361] [Q&A]
I am trying to finetune the RT-DETR for my own custom dataset. 
My dataset folder is like this:
1. Root folder is vehicle. 
2. Under the root folder 3 sub-folders are present i.e. train, val and annotations. 
3. under train folder , training images are present, 
4. under val folder validation images are present and 
5. under annotations folder train and val json file is present.
i have attached the screenshot of the yaml file 
<img width="682" alt="yaml file " src="https://github.com/ultralytics/ultralytics/assets/112607247/d10d24d6-ae78-4a3c-adf7-34d239e86f89">

I have attached the code snippet screenshot below:
<img width="1229" alt="code" src="https://github.com/ultralytics/ultralytics/assets/112607247/0465b218-b941-4529-aa2c-ec589a75db78">
When i run this code no errors are coming but in the runs folder all the metrics arte blank. nothiong is in the output . one warning message is also coming after the epoch is over saying "**WARNING  no labels found in detect set, can not compute metrics without labels**". i have attached the screenshot of the output also .
<img width="1114" alt="output" src="https://github.com/ultralytics/ultralytics/assets/112607247/23819ca6-e2af-4d06-bdb1-b2d4ab65922b">

 please help me in this . please i need to solve this immediately. please help me @glenn-jocher @pderrenger . please ... Thanks in advance

**Comment by glenn-jocher** [comment]
@arabinda-iitbbs hello,

It looks like the issue might be related to how the dataset is being read, particularly the annotations. The warning "**WARNING no labels found in detect set, can not compute metrics without labels**" suggests that the model isn't finding the label files during training.

Please ensure that your YAML file correctly points to the location of your annotation files. The paths should be relative to the YAML file itself or absolute paths from the root. Here's a quick checklist:

1. **Check Paths in YAML**: Ensure the paths for `train`, `val`, and `annotations` are correctly specified.
2. **Validate JSON Files**: Confirm that the JSON files in the `annotations` folder are correctly formatted and accessible.
3. **YAML Content**: If possible, double-check the structure of your YAML file. It should resemble:
   ```yaml
   train: vehicle/train/
   val: vehicle/val/
   nc: <number_of_classes>
   names: ['class1', 'class2', ...]
   ```

If the paths and files are correct, try a small test by manually loading the JSON files to see if they're being parsed correctly. This can help isolate the issue further.

Let us know how it goes or if you need more specific guidance!

**Reply by arabinda-iitbbs** [reply]
@glenn-jocher Thank you for the reply. I want to clarify one thing. Can we also give the dataset to RT-DETR in YOLOv8 format, the txt format? Or does RT-DETR only accept the COCO dataset format?

**Reply by glenn-jocher** [reply]
Hello @arabinda-iitbbs,

RT-DETR primarily supports the COCO dataset format for training and validation. If your dataset is currently in the YOLOv8 format (txt files), you will need to convert it to COCO format to work with RT-DETR. This ensures compatibility and allows you to utilize RT-DETR's full capabilities effectively.

If you need guidance on converting your dataset, feel free to ask!

**Comment by arabinda-iitbbs** [comment]
Thank you for your help @glenn-jocher.

**Reply by glenn-jocher** [reply]
Hello,

Thank you for reaching out and providing detailed information about your issue. It looks like you're encountering a problem with finetuning the RT-DETR model on your custom dataset, specifically related to the warning about missing labels.

Here are a few steps to help diagnose and resolve the issue:

1. **Verify Dataset Structure**:
   Ensure that your dataset structure and paths in the YAML file are correct. The `train` and `val` paths should point to the directories containing the images, and the `annotations` should correctly reference the JSON files.

2. **Check Annotations**:
   Verify that your JSON annotation files are correctly formatted and contain the necessary labels. The warning message suggests that the model is not finding any labels in your dataset, which could be due to incorrect paths or improperly formatted annotation files.

3. **YAML File**:
   Double-check your YAML file to ensure it correctly references the paths to your images and annotations. Here’s an example of how your YAML file should look:
   ```yaml
   train: /path/to/vehicle/train
   val: /path/to/vehicle/val
   nc: 1  # number of classes
   names: ['vehicle']  # class names
   ```

4. **Code Example**:
   Ensure your code for training is correctly set up. Here’s a refined example:
   ```python
   from ultralytics import RTDETR

   # Load a COCO-pretrained RT-DETR-l model
   model = RTDETR("rtdetr-l.pt")

   # Train the model on your custom dataset
   results = model.train(data="path/to/your/yaml/file.yaml", epochs=100, imgsz=640)
   ```

5. **Update Packages**:
   Make sure you are using the latest versions of `torch` and `ultralytics`. You can update them using:
   ```bash
   pip install --upgrade torch ultralytics
   ```

6. **Minimum Reproducible Example**:
   If the issue persists, please provide a minimum reproducible example of your code and dataset structure. This will help us investigate the problem more effectively. You can refer to our guide on creating a minimum reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

By following these steps, you should be able to identify and resolve the issue with your dataset and training process. If you continue to experience problems, please share additional details, and we’ll be happy to assist further.
